File: go-client-sdk/pkg/models/operations/getrulesalertsalertid.go
package operations

import (
	"github.com/speakeasy-sdks/codat-sdks/go-client-sdk/v2/pkg/models/shared"
)

type GetRulesAlertsAlertIDPathParams struct {
	AlertID string `pathParam:"style=simple,explode=false,name=alertId"`
}

type GetRulesAlertsAlertIDSecurity struct {
	APIKeyAuth *shared.SchemeAPIKeyAuth `security:"scheme,type=apiKey,subtype=header"`
	CodatLogin *shared.SchemeCodatLogin `security:"scheme,type=oauth2"`
}

type GetRulesAlertsAlertIDRequest struct {
	PathParams GetRulesAlertsAlertIDPathParams
	Security   GetRulesAlertsAlertIDSecurity
}

type GetRulesAlertsAlertIDResponse struct {
	CodatPublicAPIModelsRulesAlertModel *shared.CodatPublicAPIModelsRulesAlertModel
	ContentType                         string
	StatusCode                          int
}

File: typescript-client-sdk/src/sdk/models/operations/getrulesalertsalertid.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";
import * as shared from "../shared";


export class GetRulesAlertsAlertIdPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=alertId" })
  alertId: string;
}

export class GetRulesAlertsAlertIdSecurity extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "security, scheme=true;type=apiKey;subtype=header" })
  apiKeyAuth?: shared.SchemeAPIKeyAuth;

  @SpeakeasyMetadata({ data: "security, scheme=true;type=oauth2" })
  codatLogin?: shared.SchemeCodatLogin;
}

export class GetRulesAlertsAlertIdRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: GetRulesAlertsAlertIdPathParams;

  @SpeakeasyMetadata()
  security: GetRulesAlertsAlertIdSecurity;
}

export class GetRulesAlertsAlertIdResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  codatPublicApiModelsRulesAlertModel?: shared.CodatPublicApiModelsRulesAlertModel;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;
}